core/common/src/main/java/alluxio/uri/SingleMasterAuthority.java

```diff
@@ -63,7 +63,7 @@ public boolean equals(Object o) {
 
   @Override
   public int hashCode() {
-    return Objects.hashCode(toString());
+    return Objects.hashCode(mHost, mPort);
```

Comment:
update `equals` to match (see other equals implementations in the codebase for reference)